crates/cheatcodes/spec/src/vm.rs

```diff
@@ -1678,6 +1678,10 @@ interface Vm {
     /// Splits the given `string` into an array of strings divided by the `delimiter`.
     #[cheatcode(group = String)]
     function split(string calldata input, string calldata delimiter) external pure returns (string[] memory outputs);
+    /// Returns the index of the first occurrence of a `key` in an `input` string.
+    /// Returns `NOT_FOUND` (i.e. `type(uint256).max`) if the `key` is not found.
+    #[cheatcode(group = String)]
+    function indexOf(string memory input, string memory key) external pure returns (uint256);
```

Comment:
An alternative UX is to return `(bool found, uint256 index)`. I'm ok with either, but if we have precedent in existing cheats we should follow the same precendent. (I'm unsure if we do offhand)

Reply:
> An alternative UX is to return `(bool found, uint256 index)`. I'm ok with either, but if we have precedent in existing cheats we should follow the same precendent. (I'm unsure if we do offhand)

I think in this case the bool will be redundant because in case on Not Found we'd still have to return a uint256 also and it can't be 0 so it'd have to be uint256.max so no point in returning bool. Also this logic has been derived from [here](https://github.com/Vectorized/solady/blob/b61da94a2d275471eec03b584d70ff31881ecfae/src/utils/LibString.sol#L551-L554), so it's better to maintain consistency. lmk wdyt.

Reply:
I'm fine with this, other languages also return some sort of sentinel / max value for not found